### PR TITLE
NCW: Add Datadog

### PR DESCRIPTION
**Pull request by MatiNavCrossmint**
## Description

This is the same as from #302 . It was removed in the #330  because WC was broken when importing the SDK.  Now [this PR ](https://github.com/Paella-Labs/crossbit-main/pull/10723) fixes that

## Test plan

Create an AA Wallet, and check it logs successfully: [DD Log](https://app.datadoghq.com/logs?query=%22GET_OR_CREATE_WALLET%22%20&cols=host%2Cservice&event=AgAAAYzWlas_Ev9f0wAAAAAAAAAYAAAAAEFZeldsaUxMQUFCdy1JcHRoOUxrd2dBQQAAACQAAAAAMDE4Y2Q2OTYtMjc0OC00NWFkLTg4NmUtNmFiMWZlOWJhMWEw&index=%2A&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&view=spans&viz=stream&from_ts=1704320833079&to_ts=1704407233079&live=true)

### CODE REVIEW

File: packages/client/wallets/aa/src/services/logging/DatadogProvider.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+type DatadogContext = object | undefined;
+
+export class DatadogProvider implements BrowserLoggerInterface {
+    logInfo(message: string, context?: unknown) {
+        const _context = context as DatadogContext;
+        init();
+        datadogLogs.logger.info(message, _context);
+    }
+    logError(message: string, context?: unknown) {
```
Comment (afeight) [inline packages/client/wallets/aa/src/services/logging/DatadogProvider.ts:15]: super nit: space btwn functions
```suggestion
    }
    
    logError(message: string, context?: unknown) {
```

File: packages/client/wallets/aa/src/services/logging/DatadogProvider.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        const _context = context as DatadogContext;
+        init();
+        datadogLogs.logger.error(message, _context);
+    }
+    logWarn(message: string, context?: unknown) {
```
Comment (afeight) [inline packages/client/wallets/aa/src/services/logging/DatadogProvider.ts:20]:
```suggestion
    }
    
    logWarn(message: string, context?: unknown) {
```

File: packages/client/wallets/aa/src/services/logging/DatadogProvider.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        const _context = context as DatadogContext;
+        init();
+        datadogLogs.logger.warn(message, _context);
+    }
+}
```
Comment (afeight) [inline packages/client/wallets/aa/src/services/logging/DatadogProvider.ts:25]: we've got a lot of repetition between these functions, can we instead use the constructor?
```
export class DatadogProvider implements BrowserLoggerInterface {
  constructor() {
    super();
    init();
  }
  
  logError(message: string, context?: unknown) {
        datadogLogs.logger.error(message, context as DatadogContext);
  }
  
  ...

File: packages/client/wallets/aa/src/services/logging/DatadogProvider.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { datadogLogs } from "@datadog/browser-logs";
+
+
```
Comment (afeight) [inline packages/client/wallets/aa/src/services/logging/DatadogProvider.ts:3]: double space–I think prettier is not running

**Review by afeight — APPROVED** [review]
looks good! Just minor comments

File: packages/client/wallets/aa/src/services/logging/DatadogProvider.ts (hunk 4 above)
Comment (MatiNavCrossmint) [reply packages/client/wallets/aa/src/services/logging/DatadogProvider.ts:3]: I just copy and paste the file. I think this rule isn't set. Anyway, the suggestions make sense to me!

**Comment by MatiNavCrossmint** [comment]
As #10723 is merged, this is good to go

**Pull request merged**